xapian-core/include/xapian/stem.h

```diff
@@ -171,6 +173,122 @@ class XAPIAN_VISIBILITY_DEFAULT Stem {
     }
 };
 
+class XAPIAN_VISIBILITY_DEFAULT KrovetzStemmer : public StemImplementation {
```

Comment:
I don't think this should be a public API class - instead put it inside the library and make the existing `Stem` constructor which takes a string handle "krovetz" by setting `internal` to a new `KrovetzStemmer` object.

That provides a consistent API and hides the implementation details away (which means issues which need changes to the implementation to solve can be solved without affecting the library's ABI).